capstone_docs/Skyrsla.Rmd

```diff
@@ -0,0 +1,584 @@
+
+```
+
+Ef enginn leikari finnst í movie_info, sækir það fyrstu fimm nöfnin úr actors dálkinum. Þetta tryggir að aðalleikarar séu skráðir, jafnvel þótt þeir finnist ekki í movie_info.
+
+```{r, eval=FALSE}
+else:
+    actors_list = movie_dict["actors"].split(",")
+    first_five_actors = [actor.strip() for actor in actors_list[:5]]
+    for actor in first_five_actors:
+        if actor:
+            movie_dict["actor_in_info"] = actor
+            cursor.execute(f"""
+                INSERT INTO main_actors_in_info ({', '.join(insert_columns)})
+                VALUES ({', '.join(['?' for _ in insert_columns])})
+            """, tuple(movie_dict[col] if col != 'actor_in_info' else actor for col in insert_columns))
+
+```
+
+Kate Winslet og Leonardo Dicaprio voru ekki nefnd í movie_info fyrir myndirnar Romeo+Juliet, Triple 9 og Christmas Carol, þó að þau höfðu verið í aðalhlutverki.  Þessum upplýsingum var því bætt við í töfluna með handvirkum hætti (e. hard-coding). Þessi aðferð er almennt ekki hentug þegar unnið er með stór gagnasöfn, þar sem hún getur verið tímafrek og óhagkvæm. Hins vegar, þar sem aðeins var um örfá gildi var að ræða, þótti þessi lausn vera nægileg.
+
+Gagnavinnsla þessa verkefnis beindist að því að skapa skýran og aðgengilegan gagnagrunn sem gerir frekari greiningu og úrvinnslu mögulega. Með notkun SQL, Python og Regex tókst að umbreyta óskipulögðum gögnum í vel skipulagt gagnasafn með áherslu á Leonardo DiCaprio og Kate Winslet. Þetta tryggði að öll gögn væru hrein, rétt flokkuð og auðvelt væri að nálgast nauðsynlegar upplýsingar til greiningar. Þessi grunnur gerði það mögulegt að framkvæma ítarlegar greiningar, s.s. samanburð á kvikmyndum, leikurum og efnisflokkun, auk þess að draga fram áhugaverðar upplýsingar á notendavænan hátt. 
+
+
+
+## 3.3 Myndræn framsetning
+
+Við gerð mælaborðsins var lögð áhersla á að hanna gröf sem svara rannsóknarspurningum okkar. Til að ná því markmiði ákváðum við að einbeita okkur að því að búa til gagnvirk gröf og tengslanet sem sýna lykilatriði úr gagnasafninu á skýran og aðgengilegan hátt.
+Við nýttum Rstudio og shiny til að búa til mælaborðið. Shiny gerði okkur kleift að þróa notendavænt og gagnvirkt umhverfi þar sem hægt er að skoða gögnin frá mismunandi sjónarhornum. Gögnin voru geymd í SQLite-gagnagrunni og síðan var notað DBI og RSQLite til að sækja þau inn í R. Með ggplot2 og plotly voru gögnin síðan framsett í formi myndrita, línurita, súlna og punkta til að auðvelda samanburð og greiningu. Við hönnun myndrita var lögð áhersla á að þau væru einföld og læsanleg.
+Einnig var hannað tengslanet sem sýnir í hvaða kvikmyndum leikarar hafa leikið saman í aðalhlutverki. Til þess nýttum við gögn úr töflunni `main_actors_in_info og unnum þau með Python og bókasafninu `NetworkX`. Tengslanetið var síðan sett fram með `Dash`, þar sem notendur geta valið hnúta milli tveggja leikara og séð hvaða kvikmyndir þeir hafa unnið saman að.
```

Comment:
Þetta lookar ekki rétt i html  `main_actors_in_info og unnum þau með Python og bókasafninu `NetworkX`. Tengslanetið var síðan sett fram með `Dash`,

Reply:
búin að laga